qcfractal/tests/test_collections.py

```diff
@@ -969,6 +970,68 @@ def test_s22_view_identical(s22_fixture):
     assert_view_identical(ds)
 
 
+def test_view_download_remote(s22_fixture):
+    client, ds = s22_fixture
+
+    ds.data.__dict__["view_url"] = "https://github.com/mattwelborn/QCArchiveViews/raw/master/S22/latest.hdf5"
+    ds.download()  # 700 kb
+
+
+def test_view_download_mock(gradient_dataset_fixture, tmp_path_factory):
+    try:
+        import requests_mock
+    except ImportError:
+        pytest.skip("Missing request_mock")
+
+    client, ds = gradient_dataset_fixture
+
+    with requests_mock.Mocker(real_http=True) as m:
+        path = pathlib.Path(tmp_path_factory.mktemp('test_collections'), 'ds_gradient_remote.hdf5')
+        view = ptl.collections.HDF5View(path)
+        view.write(ds)
+
+        fake_url = "https://qcarchiveviews.com/gradient_ds.h5"
+        ds.data.__dict__["view_url"] = fake_url
+        assert ds.data.id == ds.save()
+
+        m.get(fake_url, body=open(path, 'rb'))
```

Comment:
Is the straight open here safe?

Reply:
The resolved issues I can think of are:
- We wrote the file at path, so we can trust it.
- The file at path is not open elsewhere.
- The request is mocked, so we don't have to worry about MitM or other network security concerns. 

Let me know if you have a different concern.

Reply:
I mean to make sure we close the file handle. `with open(...) as handle: `. In a test so not worried, but likely worth a comment of why we are not doing that perhaps.